tests/unit/mixins/sl-input-based-test.js

```diff
@@ -0,0 +1,200 @@
+
+    assert.ok( subject );
+});
+
+test( 'Disabled, optional, readonly, and required are false by default', function( assert ) {
+    let testObject = Ember.Object.extend( mixinUnderTest );
+    let subject = testObject.create();
+
+    assert.strictEqual(
+        subject.get( 'disabled' ),
+        false,
+        'Disabled is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'optional' ),
+        false,
+        'Optional is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'readonly' ),
+        false,
+        'Readonly is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'required' ),
+        false,
+        'Required is false by default'
+    );
+});
+
+test( 'Disabled is passed as a parameter', function( assert ) {
+    let testObject = Ember.Object.extend( mixinUnderTest );
+    let subject = testObject.create({
+        disabled: true
+    });
+
+    assert.strictEqual(
+        subject.get( 'disabled' ),
+        true,
+        'Disabled was properly passed as a parameter'
+    );
+
+    assert.strictEqual(
+        subject.get( 'optional' ),
+        false,
+        'Optional is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'readonly' ),
+        false,
+        'Readonly is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'required' ),
+        false,
+        'Required is false by default'
+    );
+});
+
+
+test( 'Optional is passed as a parameter', function( assert ) {
+    let testObject = Ember.Object.extend( mixinUnderTest );
+    let subject = testObject.create({
+        optional: true
+    });
+
+    assert.strictEqual(
+        subject.get( 'optional' ),
+        true,
+        'Optional was properly passed as a parameter'
+    );
+
+    assert.strictEqual(
+        subject.get( 'disabled' ),
+        false,
+        'Disabled is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'readonly' ),
+        false,
+        'Readonly is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'required' ),
+        false,
+        'Required is false by default'
+    );
+});
+
+
+test( 'Readonly is passed as a parameter', function( assert ) {
+    let testObject = Ember.Object.extend( mixinUnderTest );
+    let subject = testObject.create({
+        readonly: true
+    });
+
+    assert.strictEqual(
+        subject.get( 'readonly' ),
+        true,
+        'Readonly was properly passed as a parameter'
+    );
+
+    assert.strictEqual(
+        subject.get( 'disabled' ),
+        false,
+        'Disabled is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'optional' ),
+        false,
+        'Optional is false by default'
+    );
+
+    assert.strictEqual(
+        subject.get( 'required' ),
+        false,
+        'Required is false by default'
+    );
+});
+
+test( 'Required is passed as a parameter', function( assert ) {

```

Comment:
This test is unnecessary as it is checking Ember functionality and not anything specific to the logic contained within this Mixin.  Any property of any name can be mixed into the object instantiation along with the mixin - this is Ember at work.